### PR TITLE
Fixed the navigation of filtered deck options activity

### LINKED ISSUE #12480: Creating a filtered deck shows the study options fragment with no reason
###### Reproduction Steps

1. Go to the deck picker
2. Tap the three dot menu > Create filtered deck
3. Tap ok on the dialog (the filtered Deck options activity will be shown)
4. Tap back

###### Expected Result

After tapping back, you should be back to the deck picker

###### Actual Result

You are on the study options screen and have to tap back again

###### Debug info
Refer to the [support page](https://ankidroid.org/docs/help.html) if you are unsure where to get the "debug info".

###### Research
*Enter an [x] character to confirm the points below:*

- [x] I have read the [support page](https://ankidroid.org/docs/help.html) and am reporting a bug or enhancement request specific to AnkiDroid
- [x] I have checked the [manual](https://ankidroid.org/docs/manual.html) and the [FAQ](https://github.com/ankidroid/Anki-Android/wiki/FAQ) and could not find a solution to my issue
- [x] I have searched for similar existing issues here and on the user forum
- [x] (Optional) I have confirmed the issue is not resolved in the latest alpha release ([instructions](https://docs.ankidroid.org/manual.html#betaTesting))

### PR DESCRIPTION
**Pull request by pratyaksh1610**
## Purpose / Description
Fixed the navigation of filtered deck options activity, after tapping it goes back to the deck picker.

## Fixes
Fixes #12480

## Checklist
_Please, go through these checks before submitting the PR._

- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code

https://user-images.githubusercontent.com/76740999/194762985-650f58cb-915a-4a00-bc17-edaa9b6372ad.mp4

### CODE REVIEW

**Comment by pratyaksh1610** [comment]
@BrayanDSO PTAL. Thanks.

**Comment by BrayanDSO** [comment]
Here you are clearing all the opened activities and putting the deck picker behind the filtered deck options screen.

Ideally, the study options screen should not be shown at all, and that's it

**Comment by pratyaksh1610** [comment]
@BrayanDSO I have made the change. 
PTAL. Thanks.

**Comment by pratyaksh1610** [comment]
@BrayanDSO Why these checks are failing ?

File: AnkiDroid/src/main/java/com/ichi2/anki/FilteredDeckOptions.kt [hunk 1]
```diff
@@ -281,7 +281,7 @@ class FilteredDeckOptions :
             }
         }
         finish()
-        slide(this, ActivityTransitionAnimation.Direction.FADE)
+        slide(this, ActivityTransitionAnimation.Direction.END)
```
Comment (BrayanDSO) [inline AnkiDroid/src/main/java/com/ichi2/anki/FilteredDeckOptions.kt:284]: Unnecessary change

File: AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt [hunk 2]
```diff
@@ -2002,7 +2002,7 @@ open class DeckPicker :
         } else {
             val intent = Intent()
             intent.putExtra("withDeckOptions", withDeckOptions)
-            intent.setClass(this, StudyOptionsActivity::class.java)
+            intent.setClass(this, FilteredDeckOptions::class.java)
```
Comment (BrayanDSO) [inline AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt:2005]: This breaks other functionalities, as startActicityForResult below

**Review by BrayanDSO — CHANGES_REQUESTED** [review]

File: AnkiDroid/src/main/java/com/ichi2/anki/FilteredDeckOptions.kt (hunk 1 above)
Comment (pratyaksh1610) [reply AnkiDroid/src/main/java/com/ichi2/anki/FilteredDeckOptions.kt]: @BrayanDSO I have made the change.

File: AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt (hunk 2 above)
Comment (pratyaksh1610) [reply AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt]: Fixed the navigation

Comment (pratyaksh1610) [reply AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt]: Updated the recording in the PR comment.

**Comment by mikehardy** [comment]
> [Merge branch 'ankidroid:main' into branch-12480](https://github.com/ankidroid/Anki-Android/pull/12506/commits/072b215d984255dd462cd1c89192580103c1bd3b)

Why the merge commit?  https://github.com/ankidroid/Anki-Android/wiki/Development-Guide#git-workflow

**Comment by pratyaksh1610** [comment]
@mikehardy Did this so as to sync the branch

**Comment by mikehardy** [comment]
Read the link I posted. Notice the word "rebase" in the flow...

**Comment by pratyaksh1610** [comment]
PTAL @BrayanDSO

File: AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt [hunk 3]
```diff
@@ -2001,9 +2001,14 @@ open class DeckPicker :
             loadStudyOptionsFragment(withDeckOptions)
         } else {
             val intent = Intent()
-            intent.putExtra("withDeckOptions", withDeckOptions)
-            intent.setClass(this, StudyOptionsActivity::class.java)
-            startActivityForResultWithAnimation(intent, SHOW_STUDYOPTIONS, START)
+            if (withDeckOptions) {
+                intent.setClass(this, FilteredDeckOptions::class.java)
+                startActivityWithAnimation(intent, START)
+            } else {
```
Comment (BrayanDSO) [inline AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt:2007]: Ideally, the method called `openStudyOptions` should only open the study options fragment, respecting the princple of the least surprise. 

So, the filtered deck options opening should be rerouted into another method (probably a new one)

**Review by BrayanDSO — CHANGES_REQUESTED** [review]

**Comment by pratyaksh1610** [comment]
@BrayanDSO I have created a separate method. PTAL.

File: AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt (hunk 3 above)
Comment (pratyaksh1610) [reply AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt]: Done

**Comment by pratyaksh1610** [comment]
PTAL @BrayanDSO

**Comment by BrayanDSO** [comment]
Hi. Sorry, but we are all volunteers here, and I'm quite busy at the moment. Repinging me won't help at all, so I'd like if you would be patient and just wait and some day this will get reviewed. Insisting on repinging would actually reduce any will of mine to review this.

**Comment by pratyaksh1610** [comment]
Okay, cool , no worries.

**Comment by pratyaksh1610** [comment]
I think it's been a long while now, would like to know the status of the PR if possible. Thanks !

**Comment by pratyaksh1610** [comment]
@BrayanDSO PTAL.

**Review by david-allison — APPROVED** [review]
Tested on tablet and mobile. Works.
Rebased on `main` and it didn't break lint, but we should be careful here.

@pratyaksh1610 If possible, could you `squash` this into 1 commit and `rebase` it onto `main` so we can run this against our current CI

**Review by mikehardy — APPROVED** [review]
OK

**Pull request merged**